kowalski/alert_broker_ztf.py

```diff
@@ -849,11 +850,70 @@ def __init__(self, **kwargs):
 
         # todo: fetch user-defined filters, set up watchdog for near-live updates
 
-        # load *active* user-defined alert filter templates
-        active_filters = list(
+        # load *active* user-defined alert filter templates and pre-populate them
+        active_filters = self.get_active_filters()
+        self.filter_templates = self.make_filter_templates(active_filters)
+
+        self.filter_monitor = threading.Thread(target=self.reload_filters)
+        self.filter_monitor.start()
+
+        log("Loaded user-defined filters:")
+        log(self.filter_templates)
+
+    def api_skyportal(self, method: str, endpoint: str, data=None):
+        """Make an API call to a SkyPortal instance
+
+        :param method:
+        :param endpoint:
+        :param data:
+        :return:
+        """
+        method = method.lower()
+        methods = {
+            "head": self.session.head,
+            "get": self.session.get,
+            "post": self.session.post,
+            "put": self.session.put,
+            "patch": self.session.patch,
+            "delete": self.session.delete,
+        }
+
+        if endpoint is None:
+            raise ValueError("Endpoint not specified")
+        if method not in ["head", "get", "post", "put", "patch", "delete"]:
+            raise ValueError(f"Unsupported method: {method}")
+
+        if method == "get":
+            response = methods[method](
+                f"{config['skyportal']['protocol']}://"
+                f"{config['skyportal']['host']}:{config['skyportal']['port']}"
+                f"{endpoint}",
+                params=data,
+                headers=self.session_headers,
+            )
+        else:
+            response = methods[method.lower()](
```

Comment:
```suggestion
            response = methods[method](
```
already lowercased earlier in the function